worlds/dark_souls_3/Locations.py

```diff
@@ -201,7 +202,7 @@ def location_groups(self) -> List[str]:
         if self.lizard: names.append("Small Crystal Lizards")
         if self.hidden: names.append("Hidden")
 
-        default_item = item_dictionary[self.default_item_name]
+        default_item = item_dictionary[cast(str, self.default_item_name)]
```

Comment:
If we change the `self.is_event` check above into an explicit check for `self.default_item_name == None`, the fact that it's not `None` here may be inferred.

Reply:
It actually gets mad about `== None`, but it also doesn't fix anything. I did just realize we don't need any of the `is_event` check here though, since it's always checked before `location_groups()` is even called.